### PR TITLE
Add stop rollup job support to HL REST Client

### PR DESCRIPTION
**Pull request by cbuescher**
This change adds support for stoping a rollup job to the High Level REST Client.

Relates to #29827

### CODE REVIEW

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@hub-cap I think this PR is largely analogous to #34623, would you mind taking a look?

**Comment by polyfractal** [comment]
Rollup stuff LGTM.  The client stuff does too, but I'll let @hub-cap cover that if he wants :)

Aside and unrelated to this PR: we should probably adjust `AcknowledgedResponse` to not use `ToXContent` in a later PR, since it is used by a number of other client APIs.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java]: `.addPathPartAsIs("_xpack", "rollup", "job")`

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java
Comment (hub-cap) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java]: `            .addPathPartAsIs("_stop")`

File: client/rest-high-level/src/test/java/org/elasticsearch/client/RollupRequestConvertersTests.java
Comment (hub-cap) [inline client/rest-high-level/src/test/java/org/elasticsearch/client/RollupRequestConvertersTests.java]: `assertNull(request.getEntity())`

**Review by hub-cap — APPROVED** [review]
super minor nits

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java
Comment (cbuescher) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RollupRequestConverters.java]: gone

**Comment by cbuescher** [comment]
@hub-cap thanks for the review, I pushed another commit adressing your comments and am waiting on the green light from CI now.

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Pull request merged**
